HelpMyStreetFE/HelpMyStreetFE/Services/AddressService.cs

```diff
@@ -79,6 +80,62 @@ public async Task<GetPostcodesResponse> GetFriendlyNames(List<string> postcodes)
 
         }
 
+        public async Task<ResponseWrapper<GetLocationsByDistanceResponse, AddressServiceErrorCode>> GetLocationsByDistance(int distance, string postcode)
+        {
+            GetLocationsByDistanceRequest getLocationsByDistanceRequest = new GetLocationsByDistanceRequest()
+            {
+                MaxDistance = distance,
+                Postcode = postcode
+            };
+            string json = JsonConvert.SerializeObject(getLocationsByDistanceRequest);
+            StringContent data = new StringContent(json, Encoding.UTF8, "application/json");
+            HttpResponseMessage response = await Client.PostAsync("/api/GetLocationsByDistance", data);
+            string str = await response.Content.ReadAsStringAsync();
+            return JsonConvert.DeserializeObject<ResponseWrapper<GetLocationsByDistanceResponse, AddressServiceErrorCode>>(str);
+        }
+
+        public async Task<ResponseWrapper<GetLocationResponse, AddressServiceErrorCode>> GetLocationDetails(Location location)
+        {
+            var locationRequest = new LocationRequest();
+            locationRequest.Location = location;
+
+            var getLocationRequest = new GetLocationRequest();
+            getLocationRequest.LocationRequest = locationRequest;
+
+            string json = JsonConvert.SerializeObject(getLocationRequest);
+            StringContent data = new StringContent(json, Encoding.UTF8, "application/json");
+            HttpResponseMessage response = await Client.PostAsync("/api/GetLocationsByDistance", data);
+            string str = await response.Content.ReadAsStringAsync();
+            return JsonConvert.DeserializeObject<ResponseWrapper<GetLocationResponse, AddressServiceErrorCode>>(str);
+        }
+
+        public async Task<List<LocationDetails>> GetLocationDetails(IEnumerable<Location> locations)
+        {
+
+            var locationRequests = locations.Select(location => {
+                var lr = new LocationRequest();
+                lr.Location = location;
+                var glr = new GetLocationRequest();
+                glr.LocationRequest = lr;
+                return glr;
+            });
+
+            var responses = new List<LocationDetails>();
+
+            foreach (GetLocationRequest glr in locationRequests) {
+                string json = JsonConvert.SerializeObject(glr);
+                StringContent data = new StringContent(json, Encoding.UTF8, "application/json");
+                HttpResponseMessage response = await Client.PostAsync("/api/GetLocationsByDistance", data);
```

Comment:
/api/GetLocationsByDistance?

Reply:
Also...  I think we would benefit from utilising the AddressRepository as well as the AddressService here.  I know some of the exiting code doesn't match this, but the model should be:
* Anything interacting directly with the external service goes in the Repository.
* Anything doing extra clever stuff goes in the Service.
* Return types from the Service should be friendly to the rest of the application, rather than tightly coupled to the external service.

So:
* GetLocationDetails(Location location) as written above would move to the Repo.
* GetLocationDetails(Location location) in the Service would call GetLocationDetails(Location location) in the Repo, and return a LocationDetails rather than a GetLocationResponse.
* GetLocationDetails(IEnumerable<Location> locations) would stay in the Service, and call GetLocationDetails(Location location) in the Service - which would then call GetLocationDetails(Location location) in the Repo

Finally, in terms of extra clever stuff to do in the Service:
* GetLocationDetails(Location location) could usefully have some caching, so we don't need to call the external service every time.  There are a few steps to setting this up, including adding a new Singleton around line 190 of Startup.cs.  A useful model could be GetGroupById in GroupService.cs.
* GetLocationDetails(IEnumerable<Location> locations) can make its calls in parallel.  The easiest way is probably via "await Task.WhenAll" - there are a few examples in our code base if you search for that.